packages/web/src/lib/web-mode.ts

```diff
@@ -128,6 +129,9 @@ export class WebMode {
     }
 
     const txInfo = await this.sendTransaction(signedTxn);
-    return await this.waitForConfirmation(txInfo.txId);
+    if (typeof txInfo.txId === "string") {
+      return await this.waitForConfirmation(txInfo.txId);
+    }
```

Comment:
Could you please explain this? What could be the value of `txInfo.txId` if not string?

Reply:
Please check the typings, according to algosigner types we are getting jsonpayload, which could be undefined, number, etc.

Reply:
my question is wouldn't this code break on `typeof undefined.txId` (cannot fetch property `txId` of `undefined`..). Or is `typeof undefined.key` results in `undefined` also?

Reply:
return types are of `txInfo` right (not txInfo.txId) ?

Reply:
what do you suggest ?

Reply:
if `typeof undefined.txId` will break then we should use 
`if (typeof txInfo?.txId === "string")` OR
`if (txInfo && typeof txInfo.txId === "string")`

Note: we'll need to use the latter, as we can't use optional chaning. But my comment was really a question, i am not sure myself. Will need to test if it will break or not.